tests/rptest/tests/partition_movement_test.py

```diff
@@ -62,12 +76,17 @@ def __init__(self, ctx, *args, **kwargs):
             **kwargs)
         self._ctx = ctx
 
-    @cluster(num_nodes=3)
-    def test_moving_not_fully_initialized_partition(self):
+    @cluster(num_nodes=3, log_allow_list=PREV_VERSION_LOG_ALLOW_LIST)
+    @matrix(num_to_upgrade=[0, 2])
+    def test_moving_not_fully_initialized_partition(self, num_to_upgrade):
         """
         Move partition before first leader is elected
         """
-        self.start_redpanda(num_nodes=3)
+        test_mixed_versions = num_to_upgrade > 0
+        install_opts = InstallOptions(
+            install_previous_version=test_mixed_versions,
+            num_to_upgrade=num_to_upgrade)
+        self.start_redpanda(num_nodes=3, install_opts=install_opts)
 
```

Comment:
do i understand correctly that when num_to_upgrade=0 the test behaves exactly as before, and when it is 2 the test runs in a mixed version context?

Reply:
Yeah that's right. I'll add a comment to that effect.